test/e2e/certificate_publisher_test.go

```diff
@@ -21,69 +23,39 @@ import (
 // references a secret that does not exist, then creates the secret and verifies
 // that the operator updates the "router-certs" global secret.
 func TestCreateIngressControllerThenSecret(t *testing.T) {
-	cl, ns, err := getClient()
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	name := names.SimpleNameGenerator.GenerateName("test-")
-
-	// Create the ingresscontroller.
-	var one int32 = 1
-	ic := &operatorv1.IngressController{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: ns,
-		},
-		Spec: operatorv1.IngressControllerSpec{
-			DefaultCertificate: &corev1.LocalObjectReference{
-				Name: name,
-			},
-			Domain: name,
-			EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
-				Type: operatorv1.PrivateStrategyType,
-			},
-			Replicas: &one,
-		},
+	name := types.NamespacedName{Namespace: operatorNamespace, Name: names.SimpleNameGenerator.GenerateName("test-")}
+	ic := newPrivateController(name, name.Name+"."+dnsConfig.Spec.BaseDomain)
+	ic.Spec.DefaultCertificate = &corev1.LocalObjectReference{
+		Name: name.Name,
 	}
-	if err := cl.Create(context.TODO(), ic); err != nil {
-		t.Fatalf("failed to create the ingresscontroller: %v", err)
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
 	}
-	defer func() {
-		if err := cl.Delete(context.TODO(), ic); err != nil {
-			t.Fatalf("failed to delete the ingresscontroller: %v", err)
-		}
-	}()
+	defer assertIngressControllerDeleted(t, kclient, ic)
 
-	// Wait for the ingresscontroller to be reconciled.
-	err = wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
-		if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: ic.Namespace, Name: ic.Name}, ic); err != nil {
-			return false, nil
-		}
-		if len(ic.Status.Domain) == 0 {
-			return false, nil
-		}
-		return true, nil
-	})
+	conditions := []operatorv1.OperatorCondition{
+		{Type: iov1.IngressControllerAdmittedConditionType, Status: operatorv1.ConditionTrue},
+	}
```

Comment:
It might be worth introducing a named constant for this slice since you repeat it a couple times.